### PR TITLE
[SYCLomatic] Use helper function to migrate cublas<T>gemmBatched, cublas<T>trsmBatched and cublas<T>trmm

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Comment by zhiweij1** [comment]
Test case: https://github.com/oneapi-src/SYCLomatic-test/pull/35

**Review by zhimingwang36 — APPROVED** [review]
LGTM  (ask Yury to review the new added helper functions.)

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp [hunk 1]
```diff
@@ -385,6 +385,56 @@ inline void rk_impl(cl::sycl::queue &q, oneapi::mkl::uplo uplo,
...
+      reinterpret_cast<Tb **>(b), mat_info->ld_info + 1, 1,
+      &(mat_info->groupsize_info));
+
+  q.submit([&](cl::sycl::handler &cgh) {
+    cgh.depends_on(e);
+    cgh.host_task([=] { delete mat_info; });
+  });
+}
+
 }
```
Comment (YuriPlyakhin) [inline clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: I guess, this is the ending of detail namespace. If so, please add a comment "// detail" just to improve readability a bit.

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp [hunk 2]
```diff
@@ -385,6 +385,56 @@ inline void rk_impl(cl::sycl::queue &q, oneapi::mkl::uplo uplo,
   }
 }
 
+template <class Ta, class Tb, class Ts>
+inline void
+trsm_batch_impl(cl::sycl::queue &q, oneapi::mkl::side left_right,
+                oneapi::mkl::uplo upper_lower, oneapi::mkl::transpose trans,
+                oneapi::mkl::diag unit_diag, int m, int n, const void *alpha,
+                const void **a, int lda, void **b, int ldb, int batch_size) {
+  struct mat_info_t {
```
Comment (YuriPlyakhin) [inline clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: maybe: math_info_t  ? Not sure, what "mat" means.

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp [hunk 3]
```diff
@@ -1367,5 +1417,109 @@ inline void herk(cl::sycl::queue &q, oneapi::mkl::uplo uplo,
                                         ldb, beta, c, ldc);
 }
 
+/// This routines perform a group of trsm operations. Each trsm solves an
```
Comment (YuriPlyakhin) [inline clang/test/dpct/helper_files_ref/include/blas_utils.hpp]:
```suggestion
/// This routine performs a group of trsm operations. Each trsm solves an
```

**Review by YuriPlyakhin — APPROVED** [review]
just a few minor comments from my side.

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp (hunk 2 above)
Comment (zhiweij1) [reply clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: "mat" means "matrix"

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp (hunk 1 above)
Comment (zhiweij1) [reply clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: OK. I will do that.

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp (hunk 3 above)
Comment (zhiweij1) [reply clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: Will fix it

File: clang/test/dpct/helper_files_ref/include/blas_utils.hpp (hunk 2 above)
Comment (YuriPlyakhin) [reply clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: If it is not common abbreviation/short form, then may be use the full form, like matrix_info_t

Comment (zhiweij1) [reply clang/test/dpct/helper_files_ref/include/blas_utils.hpp]: OK

**Review by ziranzha — APPROVED** [review]
LGTM

**Pull request merged**